src/main/java/com/iexec/core/task/TaskController.java

```diff
@@ -137,10 +127,19 @@ ReplicateModel buildReplicateModel(Replicate replicate) {
     })
     public ResponseEntity<TaskLogs> getTaskLogs(
```

Comment:
Do you think `getTaskLogs` and `getComputeLogs` could be refactored in a single method? Something like:
```java
private <T> ResponseEntity<T> getLogs(
        String chainTaskId,
        String authorization,
        Supplier<T> logsGetter) {
    SignedChallenge signedChallenge = SignedChallenge.createFromString(authorization);
    if (signedChallenge == null) {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
    }
    String logsRequester = signedChallenge.getWalletAddress();
    if (!isTaskRequester(logsRequester, chainTaskId)) {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
    }
    Signature signature = new Signature(Numeric.cleanHexPrefix(signedChallenge.getChallengeSignature()));
    EIP712Challenge eip712Challenge = challengeService.getChallenge(logsRequester);
    if (!SignatureUtils.doesSignatureMatchesAddress(
            signature.getR(), signature.getS(), eip712Challenge.getHash(), logsRequester)) {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
    }
    return logsGetter.get();
}
```

This could then be used like this:
```java
public ResponseEntity<ComputeLogs> getComputeLogs(
        @PathVariable("chainTaskId") String chainTaskId,
        @PathVariable("walletAddress") String walletAddress,
        @RequestHeader("Authorization") String authorization) {
    return getLogs(chainTaskId, authorization, () -> taskLogsService.getComputeLogs(chainTaskId, walletAddress));
}
```

It would avoid some duplicates.

Reply:
I agree with you and am pretty sure, this will need rework at a later date.
I suggest to handle this in a next PR if you agree.

Reply:
Sure, thanks!